### ISSUE TITLE
Markdown not displaying correctly from Firestore

### ISSUE THREAD

**Issue by alexandrumic** [iamacup/react-native-markdown-display#110]
Hello, 

I am using this package to display a markdown from a collection stored in Firebase - Firestore database.

I need to store the markup as a string there, so the value of it is the following: 

"## Hello  ### This is a markdown file  **How are you doing?**  *Good, you?*   ~~This is a Strikethrough~~   #### Unordered list  + Create a list by starting a line with `+`, `-`, or `*` + Sub-lists are made by indenting 2 spaces:     - Marker character change forces new list start:       * Ac tristique libero volutpat at       + Facilisis in pretium nisl aliquet       - Nulla volutpat aliquam velit + Very easy!  #### Ordered list 1. Lorem ipsum dolor sit amet 2. Consectetur adipiscing elit 3. Integer molestie lorem at massa  ![Minion](https://octodex.github.com/images/minion.png)"

```
## Hello  ### This is a markdown file  **How are you doing?**  *Good, you?*   ~~This is a Strikethrough~~   #### Unordered list  + Create a list by starting a line with `+`, `-`, or `*` + Sub-lists are made by indenting 2 spaces:     - Marker character change forces new list start:       * Ac tristique libero volutpat at       + Facilisis in pretium nisl aliquet       - Nulla volutpat aliquam velit + Very easy!  #### Ordered list 1. Lorem ipsum dolor sit amet 2. Consectetur adipiscing elit 3. Integer molestie lorem at massa  ![Minion](https://octodex.github.com/images/minion.png)
```

The issue is that in Javascript, the new lines are not working anymore.

I have tried to replace new lines with `\n` or `<br>` and parse them in JS but without luck.

Do you have any ideas on how can I store markdown as string in Firestore and display it correctly in the app?

Screenshot attached on how is displayed right now:
![image](https://user-images.githubusercontent.com/49684634/93607148-9f0afd00-f9d1-11ea-89f0-bd878d80842d.png)

Thank you!

**Comment by JamieBradders** [comment]
Hey @alexandrumic I have EXACTLY the same problem 😂 trying to fetch a markdown string from firestore but unable to get line breaks to work. I found that using a string from a local variable such as:

```jsx
const content = `# My title\nThis is the body of the page.`;
```

Worked fine, I'm sorry to say that I've not found a workaround yet either - I've also tried your approach by replacing `\n` with `<br/>` but no such luck. I don't suppose you got any further forward yourself or found an alternative approach?

**Comment by JamieBradders** [comment]
Alright! I found a solution. This isn't a problem with the library, but rather the way Firestore is storing the text.

Taken from the first answer [here](https://stackoverflow.com/questions/53827739/firebase-firestore-new-line-command)

>In most programming languages if you include \n in a string, it interprets the two characters as a (escape) sequence and it actually stores a single non-printable character with ASCII code 10.
If you literally type \n into a document in the Firestore console, it stores that exact literal value in the string. So that's two characters \ and n.
These two are not the same. In fact, if you'd want to enter the two-character sequence in a string in most programming languages you'd end up with "\\n". That's two backslashes: the first to start an escape sequence, the second to indicate it's a literal \, and then a literal n.
So in Flutter, if you've stored the literal two-characters \n in a string and you want to display it as a newline, you need to decode the string back into a single line break character. This is luckily quite simple with:

So for me I had to do:

```jsx
<Markdown>
  {pages[index].data().body.replaceAll('\\n', '\n')}
</Markdown>
```

Then Job's a Good'un! I can't close this issue but I'd suggest closing it now 🙂 👍

**Comment by alexandrumic** [comment]
Hey @JamieBradders thank for the answer.

I have tried that solution too but didn't worked for me because I have pretty complex messages (with titles, subtitles, bolds, italics, links, images etc) and in some situations with that solution I ended up with an `Strings must be wrapped in a Text Component` error or by not displaying rest of titles, bolds etc. (things like **test** or ##Hello) correctly after the first paragraph.

What I have ended up doing and seems that is working ok so far, but it's not the most optimal solution, is to split the body by `\n` character and basically to create multiple `<Markdown>` components and put them together in a render function that I'm calling inside a `View`:

```typescript
const renderMessages = (message: string): JSX.Element[] | null => {
  try {
    const messagesElement: JSX.Element[] = [];
    const messagesArr = message.split('\\n');

    messagesArr.forEach((part, index) => {
      messagesElement.push(
        <Markdown key={index} onLinkPress={(url: string) => onLinkPress(url)}>
          {part.trim()}
        </Markdown>
      );
    });

    return messagesElement;
  } catch {
    return null;
  }
};
```
